port/cpl_recode_iconv.cpp

```diff
@@ -256,7 +256,7 @@ char *CPLRecodeFromWCharIconv( const wchar_t *pwszSource,
         reinterpret_cast<char*>(pszIconvSrcBuf));
 
     /* iconv expects a number of bytes, not characters */
-    nSrcLen *= sizeof(wchar_t);
+    nSrcLen *= nTargetCharWidth;
```

Comment:
I believe that lines 280 and 281 should also be modified to:
```
                nSrcLen -= nTargetCharWidth;
                pszSrcBuf += nTargetCharWidth;
```

Reply:
Good catch. There is also the corner case that the destination buffer might already be full when the terminating null character is written.